pkg/providers/instancetype/instancetype.go

```diff
@@ -148,6 +149,14 @@ func (p *Provider) createOfferings(ctx context.Context, instanceType *ec2.Instan
 }
 
 func (p *Provider) getInstanceTypeZones(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) (map[string]sets.String, error) {
+	// DO NOT REMOVE THIS LOCK ----------------------------------------------------------------------------
+	// We lock here so that multiple callers to getInstanceTypeZones do not result in cache misses and multiple
+	// calls to EC2 when we could have just made one call. This lock is here because multiple callers to EC2 result
+	// in A LOT of extra memory generated from the response for simultaneous callers.
+	// TODO @joinnis: This can be made more efficient by holding a Read lock and only obtaining the Write if not in cache
+	p.zoneMu.Lock()
```

Comment:
does separating the mutexes meaningfully improve performance? Alternatively, you could just lock at the front door to keep things simple.